src/main/java/net/snowflake/ingest/utils/Utils.java

```diff
@@ -132,7 +132,9 @@ public static Properties createProperties(Properties inputProp) {
     }
 
     if (!properties.containsKey(Constants.ROLE)) {
-      throw new SFException(ErrorCode.MISSING_CONFIG, "role");
+      logger.logInfo(
+          "Missing user role property in the configuration file, the default user role will be"
```

Comment:
```suggestion
          "Snowflake role is not provided, user's default role will be used"
```

It's not always we use configuration file (For making our lives easier, testing uses this profile.json)